File: Competetive programming/C++/Unit 4/Bridge.cpp
#include <iostream>
using namespace std;

int main() {

    int a, b, c, d;

    cout << "===== BRIDGE PROBLEM =====\n\n";

    cout << "Enter crossing times in ascending order.\n";
    cout << "Example: 1 2 5 10\n\n";

    cout << "Person 1 time: ";
    cin >> a;

    cout << "Person 2 time: ";
    cin >> b;

    cout << "Person 3 time: ";
    cin >> c;

    cout << "Person 4 time: ";
    cin >> d;

    int method1 = a + 2*b + d;
    int method2 = 2*a + c + d;

    int answer = min(method1, method2);

    cout << "\nMinimum Time Required = " << answer << endl;

    return 0;
}

File: Competetive programming/Python/Unit 4/Bridge.py
print("===== BRIDGE =====\n")

n = int(input("Enter number of people: "))

times = []

for i in range(n):

    value = int(input(f"Time of Person {i+1}: "))

    times.append(value)

times.sort()

total = sum(times)

print("\nCrossing Times:")
print(times)

print("\nSimple Total Time =", total)